diff --git a/os/Makefile b/os/Makefile
index 2971854..50a88f1 100644
--- a/os/Makefile
+++ b/os/Makefile
@@ -30,7 +30,7 @@ SRCS-y += $(shell echo $(SELF_DIR)/src/**/*.c)
 #CFLAGS += -I $(RTE_SDK)/$(RTE_TARGET)/include --include rte_config.h
 
 # Build with the libnf
-LDFLAGS += $(SELF_DIR)/../$(NF)/libnf.so
+SRCS-y += $(shell echo "$(SELF_DIR)/../$(NF)/"*.c)
 
 # Link with DPDK
 #LDFLAGS += -L $(RTE_SDK)/$(RTE_TARGET)/lib -ldpdk
