diff --git a/common/parsing_helpers.h b/common/parsing_helpers.h
index fb819d2..d733b2d 100644
--- a/common/parsing_helpers.h
+++ b/common/parsing_helpers.h
@@ -142,7 +142,7 @@ static __always_inline int parse_iphdr(struct hdr_cursor *nh,
 
 	hdrsize = iph->ihl * 4;
 	/* Sanity check packet field is valid */
-	if(hdrsize < sizeof(iph))
+	if(hdrsize < sizeof(*iph))
 		return -1;
 
 	/* Variable-length IPv4 header, need to use byte-based arithmetic */
@@ -238,7 +238,7 @@ static __always_inline int parse_tcphdr(struct hdr_cursor *nh,
 
 	len = h->doff * 4;
 	/* Sanity check packet field is valid */
-	if(len < sizeof(h))
+	if(len < sizeof(*h))
 		return -1;
 
 	/* Variable-length TCP header, need to use byte-based arithmetic */
